src/repositories/admin/domain-blocks-repository.ts

```diff
@@ -0,0 +1,95 @@
+import type { MastoConfig } from '../../config';
+import { version } from '../../decorators';
+import type { Admin } from '../../entities';
+import type { SeverityType } from '../../entities/admin/domain-block';
+import type { Http } from '../../http';
+
+export interface AdminBlockDomainParams {
+  /** The domain to block federation required*/
+  readonly domain: string;
+  /** Whether to apply a silence, suspend, or noop to the domain?*/
+  readonly severity?: SeverityType;
+  /** Whether media attachments should be rejected*/
+  readonly reject_media?: boolean;
+  /** Whether reports from this domain should be rejected*/
+  readonly reject_reports?: boolean;
+  /**  A private note about this domain block, visible only to admins*/
+  readonly private_comment?: boolean | null;
+  /** A public note about this domain block, optionally shown on the about page*/
+  readonly public_comment?: string | null;
+  /** Whether to partially censor the domain when shown in public*/
+  readonly obfuscate?: boolean;
+}
+
+export type AllBlockedDomain = {
```

Comment:
I could not understand what `AllBlockedDomain` is for by its name. It should be `FetchAllBlockedDomainParams` or something like that.